core/embed/rust/src/ui/model_tt/layout.rs

```diff
@@ -573,6 +707,121 @@ extern "C" fn new_request_slip39(n_args: usize, args: *const Obj, kwargs: *mut M
     unsafe { util::try_with_args_and_kwargs(n_args, args, kwargs, block) }
 }
 
+extern "C" fn new_select_word(n_args: usize, args: *const Obj, kwargs: *mut Map) -> Obj {
+    let block = move |_args: &[Obj], kwargs: &Map| {
+        let title: StrBuffer = kwargs.get(Qstr::MP_QSTR_title)?.try_into()?;
+        let description: StrBuffer = kwargs.get(Qstr::MP_QSTR_description)?.try_into()?;
+        let word0: StrBuffer = kwargs.get(Qstr::MP_QSTR_word0)?.try_into()?;
```

Comment:
why not accept a list from python?

Reply:
it seems nicer to expose `List::len()` which i think doesn't currently exist? and return a ValueError for bad lengths

Reply:
Fixed in 4a0abf4f5cd0b5f740391ccdcc69a1393592f838, should work for all iterables.